gateleen-core/src/main/java/org/swisspush/gateleen/core/util/HttpHeaderUtil.java

```diff
@@ -30,4 +31,26 @@
         return headers;
     }
 
+    /**
+     * Helper method to find the first occurrence of a http header within the given List of headers.
+     *
+     * @param headers   The Map with the header key - value pairs to be evaluated
+     * @param headerKey The key for the header pair we are searching for in the given map. Note that
+     *                  the key searching is non case sensitive.
+     * @return The found header value or null if none found
+     */
+    public static <T extends MultiMap> String getHeaderValue(T headers, String headerKey) {
```

Comment:
We should annotate this with @Nullable ?

Reply:
Yes, and with Java 11 we could even as well use @Nonnull as well now